File: MacTemplet/Cagtegory/NSStatusItem+Helper.m
//
//  NSStatusItem+Helper.m
//  MacTemplet
//
//  Created by Bin Shang on 2019/6/19.
//  Copyright © 2019 Bin Shang. All rights reserved.
//

#import "NSStatusItem+Helper.h"

@implementation NSStatusItem (Helper)

+ (instancetype)createStatusItemImageName:(nullable NSString *)imageName{
    
    NSImage * image = [NSImage imageNamed:imageName];
    
    NSStatusItem *statusItem = ({
        NSStatusBar *statusBar = NSStatusBar.systemStatusBar;
        
        NSStatusItem *item = [statusBar statusItemWithLength:NSSquareStatusItemLength];
        item.button.cell.highlighted = true;
        item.button.image = image ? : NSApplication.appIcon;
        item.button.toolTip = NSApplication.appName;
        
        item;
    });
    
    return statusItem;
}


@end

File: MacTemplet/MacCagtegory/NSStatusItem+Helper.swift
//
//  NSStatusItem+Helper.swift
//  MacTemplet
//
//  Created by Bin Shang on 2019/11/23.
//  Copyright © 2019 Bin Shang. All rights reserved.
//

import Cocoa

@objc public extension NSStatusItem {

    static func create(imageName: String?) -> NSStatusItem {
        
        var image = NSApplication.appIcon.resize(CGSize(width: 17, height: 17), isPixels: true);
        if imageName != nil && (NSImage(named: imageName!) != nil) {
            image = NSImage(named: imageName!)!
        }
        
        let statusItem: NSStatusItem = {
            let item = NSStatusBar.system.statusItem(withLength: -2)
            item.button?.cell?.isHighlighted = false;
            item.button?.image = image;
            item.button?.toolTip = NSApplication.appName;
            
            return item;
        }()
        return statusItem;
    }
}
